### PR TITLE
Small refactors on the keras.utils module

### PR DESCRIPTION
**Pull request by eltonvs**
### Summary

This PR is just a small refactor on the `keras.utils` module, using python generators over list comprehensions whenever possible in order to use less memory and avoid computing things beforehand.

I also changed the string interpolation on those files to use `.format()` instead of just concatenating the strings.

### PR Overview

- [n] This PR requires new unit tests [y/n] (make sure tests are included)
- [n] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [n] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: keras/utils/vis_utils.py [hunk 1]
```diff
@@ -149,7 +149,7 @@ def model_to_dot(model,
                 inputlabels = str(layer.input_shape)
             elif hasattr(layer, 'input_shapes'):
                 inputlabels = ', '.join(
-                    [str(ishape) for ishape in layer.input_shapes])
```
Comment (gabrieldemarmiesse) [inline keras/utils/vis_utils.py]: Note that it's not the same to use `()` or `[]`. One is a generator and the other one is a list. A list uses more memory and is slower. So the change you did isn't needed.

Comment (eltonvs) [reply keras/utils/vis_utils.py]: This was the exactly change I did, actually, changing from a list to a generator to decrease the memory usage and increase performance.

Comment (gabrieldemarmiesse) [reply keras/utils/vis_utils.py]: Woops my bad

**Comment by gabrieldemarmiesse** [comment]
It seems that there were some flaky tests, not due to your pull request. I can't merge until all the tests are passing. Let's wait until the CI is done.

**Pull request merged**